src/lib389/lib389/password_plugins.py

```diff
@@ -56,6 +52,74 @@ def __init__(self, instance, dn=f'cn=SSHA,{DN_PWDSTORAGE_SCHEMES}'):
         super(SSHAPlugin, self).__init__(instance, dn)
 
 
+class PBKDF2BasePlugin(PasswordPlugin):
+    """Base class for all PBKDF2 variants"""
+    DEFAULT_ROUNDS = 70000  # Fallback default if not specified by subclass
+
+    def __init__(self, instance, dn):
+        super(PBKDF2BasePlugin, self).__init__(instance, dn)
+        self._create_objectclasses.append('pwdPBKDF2PluginConfig')
+        
+    def set_rounds(self, rounds):
+        """Set the number of rounds for PBKDF2 hashing (requires restart)
+        
+        :param rounds: Number of rounds
+        :type rounds: int
+        """
+        # Ensure the pwdPBKDF2PluginConfig objectClass is present
+        self.ensure_present('objectClass', 'pwdPBKDF2PluginConfig')
+        
+        rounds = int(rounds)
+        if rounds < 10000 or rounds > 10000000:
+            raise ValueError("PBKDF2 rounds must be between 10,000 and 10,000,000")
+        self.replace('nsslapd-pwdPBKDF2NumIterations', str(rounds))
+        
+    def get_rounds(self):
+        """Get the current number of rounds
+        
+        :param use_json: Whether to return JSON formatted output
+        :type use_json: bool
+        :returns: Current rounds setting or JSON string
+        :rtype: int
+        """
+        rounds = self.get_attr_val_int('nsslapd-pwdPBKDF2NumIterations')
+        if rounds:
+            return rounds
+        return self.DEFAULT_ROUNDS
+
+
+class PBKDF2Plugin(PBKDF2BasePlugin):
+    """PBKDF2 password storage scheme"""
+    DEFAULT_ROUNDS = 70000
+
+    def __init__(self, instance, dn=f'cn=PBKDF2,{DN_PWDSTORAGE_SCHEMES}'):
+        super(PBKDF2Plugin, self).__init__(instance, dn)
+
+
+class PBKDF2SHA1Plugin(PBKDF2BasePlugin):
+    """PBKDF2-SHA1 password storage scheme"""
+    DEFAULT_ROUNDS = 70000
+
+    def __init__(self, instance, dn=f'cn=PBKDF2-SHA1,{DN_PWDSTORAGE_SCHEMES}'):
+        super(PBKDF2SHA1Plugin, self).__init__(instance, dn)
+
+
+class PBKDF2SHA256Plugin(PBKDF2BasePlugin):
+    """PBKDF2-SHA256 password storage scheme"""
+    DEFAULT_ROUNDS = 30000
+
+    def __init__(self, instance, dn=f'cn=PBKDF2-SHA256,{DN_PWDSTORAGE_SCHEMES}'):
+        super(PBKDF2SHA256Plugin, self).__init__(instance, dn)
+
+
+class PBKDF2SHA512Plugin(PBKDF2BasePlugin):
+    """PBKDF2-SHA512 password storage scheme"""
+    DEFAULT_ROUNDS = 10000
```

Comment:
also, SHA512 is actually faster to calculate with 64 bit CPU than SHA256, so why we're making it faster still with reduced round count?

Reply:
I'll make the defaults the same for all hash functions, and the administrator will be able to tune the settings for their hardware so they can make it run in approximately 100ms.

Reply:
Okay, I tested on my VM, and it looks like `100_000` will be the closest to the `100ms` default. But, of course, the administrator should tune it to their hardware.